Client/src/Utils/NetworkService.js

```diff
@@ -0,0 +1,434 @@
+import axios from "axios";
+import { clearAuthState } from "../Features/Auth/authSlice";
+const BASE_URL = import.meta.env.VITE_APP_API_BASE_URL;
+
+class NetworkService {
+  constructor(store) {
+    this.store = store;
+    this.axiosInstance = axios.create({ baseURL: BASE_URL });
+    this.axiosInstance.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        console.error(error);
+        if (error.response && error.response.status === 401) {
+          console.log("Invalid token revoked");
+          networkService;
+        }
+        return Promise.reject(error);
+      }
+    );
+  }
```

Comment:
Adding `this.defaultHeaders` will help us to extend new methods. Most(or all) endpoints work with `Content-Type: application/json` header.